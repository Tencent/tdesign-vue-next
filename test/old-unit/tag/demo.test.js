/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/examples/tag/demos/base.vue';
import deleteVue from '@/examples/tag/demos/delete.vue';
import disabledVue from '@/examples/tag/demos/disabled.vue';
import iconVue from '@/examples/tag/demos/icon.vue';
import longTextVue from '@/examples/tag/demos/long-text.vue';
import plainVue from '@/examples/tag/demos/plain.vue';
import selectableVue from '@/examples/tag/demos/selectable.vue';
import shapeVue from '@/examples/tag/demos/shape.vue';
import sizeVue from '@/examples/tag/demos/size.vue';
import themeVue from '@/examples/tag/demos/theme.vue';

const mapper = {
  baseVue,
  deleteVue,
  disabledVue,
  iconVue,
  longTextVue,
  plainVue,
  selectableVue,
  shapeVue,
  sizeVue,
  themeVue,
};

describe('Tag', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Tag ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
