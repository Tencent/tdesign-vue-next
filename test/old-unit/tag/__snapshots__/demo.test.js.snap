// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Tag baseVue demo works fine 1`] = `
<div>
  <div
    class="tag-demo"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <a
      href="https://www.tencent.com/zh-cn"
      target="_blank"
    >
      <span
        class="t-tag t-tag--default t-tag--dark t-size-m"
      >
        <!---->
        
        超链接
        
        <!---->
      </span>
    </a>
  </div>
  <div
    class="tag-demo"
  >
    <span
      class="t-tag t-tag--primary t-tag--dark t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--success t-tag--dark t-size-m"
    >
      <!---->
      
      标签二
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--warning t-tag--dark t-size-m"
    >
      <!---->
      
      标签三
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--danger t-tag--dark t-size-m"
    >
      <!---->
      
      标签四
      
      <!---->
    </span>
  </div>
  <div
    class="tag-demo light"
  >
    <span
      class="t-tag t-tag--primary t-tag--light t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--success t-tag--light t-size-m"
    >
      <!---->
      
      标签二
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--warning t-tag--light t-size-m"
    >
      <!---->
      
      标签三
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--danger t-tag--light t-size-m"
    >
      <!---->
      
      标签四
      
      <!---->
    </span>
  </div>
  <div
    class="tag-demo"
  >
    <span
      class="t-tag t-tag--primary t-tag--outline t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--success t-tag--outline t-size-m"
    >
      <!---->
      
      标签二
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--warning t-tag--outline t-size-m"
    >
      <!---->
      
      标签三
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--danger t-tag--outline t-size-m"
    >
      <!---->
      
      标签四
      
      <!---->
    </span>
  </div>
</div>
`;

exports[`Tag Tag deleteVue demo works fine 1`] = `
<div
  class="tag-demo"
>
  <div
    class="tag-block"
  >
    <span
      class="t-tag t-tag--primary t-tag--dark t-tag--close t-size-m"
    >
      <!---->
      
      标签一
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--success t-tag--dark t-tag--close t-size-m"
    >
      <!---->
      
      标签二
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--warning t-tag--dark t-tag--close t-size-m"
    >
      <!---->
      
      标签三
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--danger t-tag--dark t-tag--close t-size-m"
    >
      <!---->
      
      标签四
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
  </div>
  <div
    class="tag-block light"
  >
    <span
      class="t-tag t-tag--primary t-tag--light t-tag--close t-size-m"
    >
      <!---->
      
      标签一
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--success t-tag--light t-tag--close t-size-m"
    >
      <!---->
      
      标签二
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--warning t-tag--light t-tag--close t-size-m"
    >
      <!---->
      
      标签三
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--danger t-tag--light t-tag--close t-size-m"
    >
      <!---->
      
      标签四
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
  </div>
  <div
    class="tag-block"
  >
    <span
      class="t-tag t-tag--primary t-tag--outline t-tag--close t-size-m"
    >
      <!---->
      
      标签一
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--success t-tag--outline t-tag--close t-size-m"
    >
      <!---->
      
      标签二
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--warning t-tag--outline t-tag--close t-size-m"
    >
      <!---->
      
      标签三
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--danger t-tag--outline t-tag--close t-size-m"
    >
      <!---->
      
      标签四
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
  </div>
  <div
    class="tag-block"
  >
    
    <span
      class="t-tag t-tag--default t-tag--dark t-tag--ellipsis t-tag--close t-size-m"
      style="max-width: 100px;"
    >
      <!---->
      <span
        class="t-tag--text"
        style="max-width: 100px;"
      >
        
        可删除标签可删除标签
        
      </span>
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--default t-tag--dark t-tag--ellipsis t-tag--close t-size-m"
      style="max-width: 100px;"
    >
      <svg
        class="t-icon t-icon-discount"
      >
        <use
          href="#t-icon-discount"
        />
      </svg>
      <span
        class="t-tag--text"
        style="max-width: 100px;"
      >
        
        可删除标签可删除标签
        
      </span>
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    <span
      class="t-tag t-tag--default t-tag--dark t-tag--close t-tag--disabled t-size-m"
    >
      <!---->
      
      可删除标签
      
      <svg
        class="t-icon t-icon-close t-tag__icon-close"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M8 8.92L11.08 12l.92-.92L8.92 8 12 4.92 11.08 4 8 7.08 4.92 4 4 4.92 7.08 8 4 11.08l.92.92L8 8.92z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
    </span>
    
  </div>
  <div
    class="tag-block editable"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m"
    >
      <!---->
      
      <svg
        class="t-icon t-icon-add"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
          fill="currentColor"
          fill-opacity="0.9"
        />
      </svg>
       添加标签 
      
      <!---->
    </span>
  </div>
</div>
`;

exports[`Tag Tag disabledVue demo works fine 1`] = `
<div
  class="tdesign-tag-disabled tag-demo"
>
  <span
    class="t-tag t-tag--check t-tag--default t-size-m t-tag--disabled check-tag"
  >
    
    失效标签
    
  </span>
  <span
    class="t-tag t-tag--default t-tag--dark t-tag--disabled t-size-m"
  >
    <!---->
    
    失效标签
    
    <!---->
  </span>
</div>
`;

exports[`Tag Tag iconVue demo works fine 1`] = `
<div
  class="tdesign-demo-block-column"
>
  <div
    class="tdesign-demo-block-row"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m"
    >
      <svg
        class="t-icon t-icon-discount"
      >
        <use
          href="#t-icon-discount"
        />
      </svg>
      
       默认标签 
      
      <!---->
    </span>
  </div>
  <div
    class="tdesign-demo-block-row"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m"
    >
      
      <svg
        class="t-icon t-icon-discount"
      >
        <use
          href="#t-icon-discount"
        />
      </svg>
      
      
       Icon插槽 
      
      <!---->
    </span>
  </div>
</div>
`;

exports[`Tag Tag longTextVue demo works fine 1`] = `
<div
  class="tdesign-tag-ellipsis"
>
  <span
    class="t-tag t-tag--default t-tag--dark t-tag--ellipsis t-size-m"
    style="max-width: 150px;"
    title="默认超八个字超长文本标签超长省略文本标签"
  >
    <!---->
    <span
      class="t-tag--text"
      style="max-width: 150px;"
    >
      
      默认超八个字超长文本标签超长省略文本标签
      
    </span>
    <!---->
  </span>
</div>
`;

exports[`Tag Tag plainVue demo works fine 1`] = `
<div
  class="tdesign-tag-outline"
>
  <span
    class="t-tag t-tag--primary t-tag--outline t-size-m"
  >
    <!---->
    
    标签一
    
    <!---->
  </span>
  <span
    class="t-tag t-tag--success t-tag--outline t-size-m"
  >
    <!---->
    
    标签二
    
    <!---->
  </span>
  <span
    class="t-tag t-tag--default t-tag--outline t-size-m"
  >
    <!---->
    
    标签三
    
    <!---->
  </span>
  <span
    class="t-tag t-tag--warning t-tag--outline t-size-m"
  >
    <!---->
    
    标签四
    
    <!---->
  </span>
  <span
    class="t-tag t-tag--danger t-tag--outline t-size-m"
  >
    <!---->
    
    标签五
    
    <!---->
  </span>
</div>
`;

exports[`Tag Tag selectableVue demo works fine 1`] = `
<div
  class="tag-demo"
>
  
  <span
    class="t-tag t-tag--check t-tag--default t-size-m t-tag--checked"
  >
    
    选中
    
  </span>
  <span
    class="t-tag t-tag--check t-tag--default t-size-m"
  >
    
    未选
    
  </span>
  <span
    class="t-tag t-tag--check t-tag--default t-size-m t-tag--disabled"
  >
    
    Disabled
    
  </span>
  
</div>
`;

exports[`Tag Tag shapeVue demo works fine 1`] = `
<div>
  <div
    class="tag-demo"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--primary t-tag--dark t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
  </div>
  <div
    class="tag-demo"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m t-tag--round"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--primary t-tag--dark t-size-m t-tag--round"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
  </div>
  <div
    class="tag-demo"
  >
    <span
      class="t-tag t-tag--default t-tag--dark t-size-m t-tag--mark"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--primary t-tag--dark t-size-m t-tag--mark"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
  </div>
</div>
`;

exports[`Tag Tag sizeVue demo works fine 1`] = `
<div
  class="tag-demo"
>
  <span
    class="t-tag t-tag--default t-tag--dark t-size-s"
  >
    <!---->
    
    小型标签
    
    <!---->
  </span>
  <span
    class="t-tag t-tag--default t-tag--dark t-size-m"
  >
    <!---->
    
    默认标签
    
    <!---->
  </span>
  <span
    class="t-tag t-tag--default t-tag--dark t-size-l"
  >
    <!---->
    
    大型标签
    
    <!---->
  </span>
</div>
`;

exports[`Tag Tag themeVue demo works fine 1`] = `
<div
  class="tdesign-tag-theme"
>
  <div
    class="tdesign-demo-block"
  >
     深色 
    <span
      class="t-tag t-tag--primary t-tag--dark t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--success t-tag--dark t-size-m"
    >
      <!---->
      
      标签二
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--warning t-tag--dark t-size-m"
    >
      <!---->
      
      标签三
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--danger t-tag--dark t-size-m"
    >
      <!---->
      
      标签四
      
      <!---->
    </span>
  </div>
  <div
    class="tdesign-demo-block light"
  >
     浅色 
    <span
      class="t-tag t-tag--primary t-tag--light t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--success t-tag--light t-size-m"
    >
      <!---->
      
      标签二
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--warning t-tag--light t-size-m"
    >
      <!---->
      
      标签三
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--danger t-tag--light t-size-m"
    >
      <!---->
      
      标签四
      
      <!---->
    </span>
  </div>
  <div
    class="tdesign-demo-block plain"
  >
     朴素 
    <span
      class="t-tag t-tag--primary t-tag--outline t-size-m"
    >
      <!---->
      
      标签一
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--success t-tag--outline t-size-m"
    >
      <!---->
      
      标签二
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--warning t-tag--outline t-size-m"
    >
      <!---->
      
      标签三
      
      <!---->
    </span>
    <span
      class="t-tag t-tag--danger t-tag--outline t-size-m"
    >
      <!---->
      
      标签四
      
      <!---->
    </span>
  </div>
</div>
`;
